core/src/main/scala/io/finch/ToService.scala

```diff
@@ -94,19 +98,33 @@ object ToService {
     def apply(
         endpoints: Endpoint[A] :: ET,
         includeDateHeader: Boolean,
-        includeServerHeader: Boolean): Service[Request, Response] = new Service[Request, Response] {
+        includeServerHeader: Boolean,
+        metrics: Metrics): Service[Request, Response] = new Service[Request, Response] {
 
       private[this] val underlying = endpoints.head.handle(respond400OnErrors)
 
       def apply(req: Request): Future[Response] = underlying(Input.fromRequest(req)) match {
-        case EndpointResult.Matched(rem, out) if rem.route.isEmpty => out.map(oa => conformHttp(
-          oa.toResponse(trA, trE),
-          req.version,
-          includeDateHeader,
-          includeServerHeader
-        )).run
-
-        case _ => tsT(endpoints.tail, includeDateHeader, includeServerHeader)(req)
+        case EndpointResult.Matched(rem, out) if rem.route.isEmpty =>
+          val path = underlying.toString
+          val start = Stopwatch.timeMillis().toFloat
+          metrics.countCall(path)
+          out.map(oa => conformHttp(
+            oa.toResponse(trA, trE),
+            req.version,
+            includeDateHeader,
+            includeServerHeader
+          )).run.onSuccess(response => {
+            val now = Stopwatch.timeMillis().toFloat
```

Comment:
Wouldn't using nano's be more performant? (My takeaway from the akka-http code, may not apply here.)

Reply:
Not exactly. It totally depends on how it's implemented in concrete JVM/OS. Usually, nano is slower